### PR TITLE
Add differentiable 2D gaussian splatter

### PR DESCRIPTION
**Pull request by saipraveenb25**
There's still a couple of minor workarounds (in particular, for workgroupUniformLoad), but otherwise it works pretty well!
Needs a minute or two to properly converge.

### CODE REVIEW

File: demos/gsplat2d-diff.slang [hunk 1]
```diff
@@ -0,0 +1,647 @@
+import playground;
+
+#define BLOB_BUFFER_SIZE 184320
```
Comment (csyonghe) [inline demos/gsplat2d-diff.slang:3]: should use static const instead #define.

**Review by csyonghe — APPROVED** [review]

**Comment by csyonghe** [comment]
There is not a demo_list.json entry to point to the new shader?

File: demos/demo-list.js [hunk 2]
```diff
@@ -2,10 +2,12 @@ var demoList = [
   { name: "Simple Print", url: "simple-print.slang" },
   { name: "Simple Image", url: "simple-image.slang" },
   { name: "Image From URL", url: "image-from-url.slang" },
+  { name: "Multi-kernel Demo", url: "multiple-kernels.slang" },
```
Comment (csyonghe) [inline demos/demo-list.js:5]: This file is not checked in.

Comment (saipraveenb25) [reply demos/demo-list.js:5]: Checked it in now!

File: demos/multiple-kernels.slang [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import playground;
+
+//! @buf: RAND(131072)
```
Comment (csyonghe) [inline demos/multiple-kernels.slang:3]: Should add comment to document the feature.

**Review by csyonghe — APPROVED** [review]

**Pull request merged**